src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -284,7 +280,7 @@ public int login(String gatewayUrl, ShardInfo shardInfo, Compression compression
         String token = authConfig.getToken();
         setStatus(Status.LOGGING_IN);
         if (token == null || token.isEmpty())
-            throw new LoginException("Provided token was null or empty!");
+            throw new InvalidTokenException("Provided token was null or empty!");
```

Comment:
@DV8FromTheWorld doesn't it make more sense to throw an IllegalArgument here?

Reply:
Imo no, we should use this exception.

Reply:
This check should have been done before hand in JDABuilder, not here.
We should move this token check to the JDABuilder and use `Checks.notEmpty` to ensure it is valid.

Reply:
Will this be done in this PR or should we make a separate PR?

Reply:
We can push this to another PR